src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -21,28 +21,38 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import org.json.simple.JSONObject;
 
 @WebServlet("/login")
 public class LoginServlet extends HttpServlet {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html");
-
     UserService userService = UserServiceFactory.getUserService();
-    if (userService.isUserLoggedIn()) {
-      String userEmail = userService.getCurrentUser().getEmail();
-      String urlToRedirectToAfterUserLogsOut = "/login";
-      String logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+    boolean loggedIn = userService.isUserLoggedIn();
+    String urlToRedirect = "/search.html";
 
-      response.getWriter().println("<p>Hello " + userEmail + "!</p>");
-      response.getWriter().println("<p>Logout <a href=\"" + logoutUrl + "\">here</a>.</p>");
+    JSONObject userDetails = new JSONObject();
+    userDetails.put("loggedIn", loggedIn);
+    if (loggedIn) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      userDetails.put("userEmail", userEmail);
+      String logoutURL = userService.createLogoutURL(urlToRedirect);
+      userDetails.put("url", logoutURL);
+      int start = userEmail.indexOf('@');
+      int end = userEmail.indexOf('.');
```

Comment:
Technically this may not work as you expect. Consider the following email: "luke.travis@mit.edu"

Reply:
This is just kind of a holder until I can get OAuth running - one of my goals for the week

Reply:
Maybe add in a comment that this should be reworked in the future then. Incidentally that fix here is pretty easy: "luke.travis@mit.edu" -> "mit.edu" -> "mit"

Reply:
good point, i'll change the code to reflect that approach